src/users/dashboards.rs

```diff
@@ -189,6 +189,11 @@ impl Dashboards {
         s.retain(|d| d.dashboard_id != Some(dashboard_id.to_string()));
     }
 
+    pub fn delete_for_user(&self, user_id: &str) {
+        let mut s = self.0.write().expect(LOCK_EXPECT);
+        s.retain(|d| d.user_id != Some(user_id.to_string()));
```

Comment:
```suggestion
        let mut dashboards = self.0.write().expect(LOCK_EXPECT);
        dashboards.retain(|d| d.user_id.is_some_and(|id| id != user_id));
```
The above should help ease some memory allocations